dev/io.openliberty.microprofile.telemetry.2.0.logging.internal/src/io/openliberty/microprofile/telemetry20/logging/internal/MpTelemetryLogMappingUtils.java

```diff
@@ -302,6 +313,88 @@ private static void mapAuditLogsToOpenTelemetry(LogRecordBuilder builder, String
         builder.setContext(Context.current());
     }
 
+    /**
+     * Maps the Access log events to the OpenTelemetry Logs Data Model.
+     *
+     * @param builder   The OpenTelemetry LogRecordBuilder, which is used to construct the LogRecord.
+     * @param eventType The type of event
+     * @param event     The object originating from logging source which contains necessary fields.
+     */
+    private static void mapAccessToOpenTelemetry(LogRecordBuilder builder, String eventType, Object event) {
+        AccessLogData accessLogData = (AccessLogData) event;
+
+        builder.setSeverity(Severity.INFO2);
+
+        // Set the body to "Empty" by default. The body will be overwritten by the requestFirstLine
+        // unless disabled in logFormat configruation
+        builder.setBody("Empty");
+
+        // Get Attributes builder to add additional Log fields
+        AttributesBuilder attributes = Attributes.builder();
+
+        List<KeyValuePair> kvpList = new ArrayList<>();
+
+        AccessLogDataFormatter[] formatters = accessLogData.getFormatters();
+
+        if (formatters[4] != null) {
+            formatters[4].populate(kvpList, accessLogData);
+        } else if (formatters[5] != null) {
+            formatters[5].populate(kvpList, accessLogData);
+        }
+
+        String key = null;
+        Object value = null;
+
+        for (Iterator<KeyValuePair> element = kvpList.iterator(); element.hasNext();) {
+            KeyValuePair next = element.next();
+            key = next.getKey();
+            value = getPairValue(next);
+
+            String formattedKey = MpTelemetryAccessEventMappingUtils.getOTelMappedAccessEventKeyName(key);
+
+            if (value != null) {
+                if (key.equals("requestProtocol")) {
+                    String[] requestProtocolSplit = value.toString().split("/");
+                    attributes.put(SemanticAttributes.NETWORK_PROTOCOL_NAME, requestProtocolSplit[0]);
+                    attributes.put(SemanticAttributes.NETWORK_PROTOCOL_VERSION, requestProtocolSplit[1]);
+                } else if (key.equals("datetime") || key.equals("accessLogDatetime")) {
+                    builder.setTimestamp(formatDateTime((String) value));
+                } else if (key.contains("requestHeader") || key.contains("responseHeader")) {
+                    attributes.put(formattedKey, (String) value);
+                } else if (key.equals("requestPort")) {
+                    attributes.put(formattedKey, Integer.parseInt((String) value));
+                } else if (key.equals("requestFirstLine")) {
+                    if (!AccessLogConfig.accessLogFieldsTelemetryConfig.equals("default")) {
+                        attributes.put(formattedKey, (String) value);
+                    }
+
+                    String accessLogMsg = accessLogData.getRequestFirstLine();
```

Comment:
Why do you retrieve the requestFirstLine again, can't you just assign it to `value`?

Reply:
No, as value can be empty and not null.